dali/sasha/saxref.cpp

```diff
@@ -585,40 +585,48 @@ class CNewXRefManagerBase
     unsigned lastlog;
     unsigned sfnum;
     unsigned fnum;
+
+    // Heartbeat mechanism for long-running directory scans
+    CriticalSection heartbeatSect;
+    time_t heartbeatStartTime;
```

Comment:
minor: better to use initialisers on the members than assignments in the constructor.